dg/commands/cluster/wait_for_image_registry.py

```diff
@@ -0,0 +1,50 @@
+#  Copyright 2021 IBM Corporation
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import logging
+import socket
+
+from time import sleep
+
+import click
+
+import dg.config.cluster_credentials_manager
+import dg.lib.click.utils
+import dg.lib.openshift
+
+from dg.utils.logging import loglevel_command
+
+logger = logging.getLogger(__name__)
+
+
+@loglevel_command(
+    context_settings=dg.lib.click.utils.create_default_map_from_dict(
+        dg.config.cluster_credentials_manager.cluster_credentials_manager.get_current_credentials()
+    )
+)
+@click.option("--route-name", default="image-registry", help="Image Registry route name")
+def wait_for_image_registry(route_name: str):
+    """Wait for the Image Registry hostname to be known"""
+
+    image_registry_hostname = dg.lib.openshift.get_image_registry_hostname(route_name)
+    is_image_registry_hostname_known = False
+
+    while not is_image_registry_hostname_known:
+        try:
+            socket.gethostbyname(image_registry_hostname)
+
+            is_image_registry_hostname_known = True
+        except socket.gaierror:
+            logger.warning(f"Host '${image_registry_hostname}' is unknown")
+            sleep(1)
```

Comment:
why is the `sleep(1)` statement needed when it's not possible to open a connection to the image registry?

Reply:
This command is meant to run within a container in an OpenShift cluster. It may take a while until the image registry hostname is known and the command only returns once it is known as only then the Cloud Pak for Data installation will succeed.